src/ring_theory/polynomial/homogeneous.lean

```diff
@@ -38,6 +40,71 @@ if all monomials occuring in `φ` have degree `n`. -/
 def is_homogeneous [comm_semiring R] (φ : mv_polynomial σ R) (n : ℕ) :=
 ∀ ⦃d⦄, coeff d φ ≠ 0 → ∑ i in d.support, d i = n
 
+variables (σ R)
+
+/-- The submodule of homogeneous `mv_polynomial`s of degree `n`. -/
+noncomputable def homogeneous_submodule [comm_semiring R] (n : ℕ) :
+  submodule R (mv_polynomial σ R) :=
+{ carrier := { x | x.is_homogeneous n },
+  smul_mem' := λ r a ha c hc, begin
+    have : coeff c (r • a) = r • coeff c a := finsupp.smul_apply _ _ _,
```

Comment:
This suggests that we are missing a `coeff_smul` lemma.

Reply:
#6742

Reply:
Thanks, I kicked it on the queue